### PR TITLE
C2-19 Creating a basic logging package

### PR DESCRIPTION
**Pull request by AnthonyRodway**
This PR correlates to ticket [C2-19](https://seng499-b01-company2.atlassian.net/browse/C2-19) and correlates to setting up a basic logging package.

### Changes
- Added a basic logging file that handles info, warning and error logs depending on the situation
- The main.go file handles the logger initialization in init(), and as long as you import `"github.com/SENG-499-Company2-B01/Backend/logger"` at the top of a file you should be able to call the logger in any file in the project like:
```
logger.Info("This is an info message")
logger.Warning("This is a warning message")
logger.Error(fmt.Errorf("This is an error message"))
```
- Create a test file to test all of the logs from the console
- Added a global variable to exit the program on error logs, this emulates the log.Fatal process that an error log will most likely replace, this can be turned off and on like (this can be seen in the test file so it does not fail if the program exits during its testing):

```
logger.SetExitOnError(false)
logger.SetExitOnError(true)
```
- Below is an example of what is looks like in the console for all three variations of logs
image

### CODE REVIEW

**Review by jakecomp — APPROVED** [review]
Great setup with the logger. I think this will save us a bunch of debugging time in the fututre

File: tests/logger_test.go [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+	"log"
+	"os"
+	"strings"
+	"testing"
+	"fmt"
+
+	"github.com/SENG-499-Company2-B01/Backend/logger"
+)
+
+func TestLogger(t *testing.T) {
```
Comment (jberryman566) [inline tests/logger_test.go]: I would separate these tests into their own for cleanliness. Ideally a test will just test 1 thing (ex: Info Message)

Comment (AnthonyRodway) [reply tests/logger_test.go]: fixed 👍

**Comment by AnthonyRodway** [comment]
Ugly commit history cause of the merge conflicts 🤷‍♂️

**Review by jberryman566 — APPROVED** [review]
LGTM! Thanks for the quick fix on those tests! Looks slick

**Pull request merged**